### PR TITLE
`Spacer`: change props override order, split types

### PR DESCRIPTION
**Pull request by ciampo**
## Description

This PR cherry-picks a few minor improvements to the `<Spacer />` component  that were part of #33055 and didn't get merged when the original PR got closed:

- refactor types to a separate file
- change props override order (from more generic to more specific) to improve composition
- Add a storybook story
- Update README

## How has this been tested?

To test those changes:
- Make sure unit tests pass
- run `npm run storybook:dev` locally, visit the `Spacer` story and play around with the knobs

## Screenshots 

Screenshot 2021-07-19 at 19 15 23

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/components/src/spacer/types.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+	paddingLeft?: SpaceInput;
+	/**
+	 * Adjusts right padding.
+	 */
+	paddingRight?: SpaceInput;
+}
+
+export interface Props extends InnerProps {
+	children?: ReactNode;
+}
```
Comment (sarayourfriend) [inline packages/components/src/spacer/types.ts]: I don't think we need the `InnerProps` separated out anymore. I believe I did that just to make it possible to have it without `children` on the styled component, but that's not necessary here. You might want to recombine them.

**Review by sarayourfriend — APPROVED** [review]
LGTM! Thanks 🚀

**Review by sarayourfriend — APPROVED** [review]

File: packages/components/src/spacer/types.ts (hunk 1 above)
Comment (ciampo) [reply packages/components/src/spacer/types.ts]: Good point! Done in https://github.com/WordPress/gutenberg/pull/33555/commits/2a0cf2bcaae967a9e18338eb8cce3a305d2de88b

**Comment by talldan** [comment]
@ciampo Would be great if you could add one of the '[Type]' labels (usually enhancement or bug) to the Component System PRs.

These affect how the changes are categorized in the changelog for Gutenberg releases—at the moment whoever compiles the changelog has to go through and manually check what each PR was when the label is missing, so this would save a lot of work.

**Comment by ciampo** [comment]
> @ciampo Would be great if you could add one of the '[Type]' labels (usually enhancement or bug) to the Component System PRs.
> 
> These affect how the changes are categorized in the changelog for Gutenberg releases—at the moment whoever compiles the changelog has to go through and manually check what each PR was when the label is missing, so this would save a lot of work.

Hey @talldan, thank you for flagging this. I've added a label, sorry if I didn't include it before!

**Pull request merged**